### DISCUSSION TITLE
Changing CsvConfiguration after CsvWriter initialization

### DISCUSSION THREAD

**Discussion by tattdogg** [JoshClose/CsvHelper#1851] [Q&A]
Is it possible to change CsvConfiguration after CsvWriter initialization?
in older versions I used this:
`using var csv = new CsvWriter(stream, culture);
csv.Configuration.PrepareHeaderForMatch = x => x.Header.Trim().ToUpper();`

now it doesn't work and I have to create csvConfiguration before writer initialization:
`var config = new CsvConfiguration(culture) {
PrepareHeaderForMatch = x => x.Header.Trim().ToUpper();
};
using var csv = new CsvWriter(stream, config);`

I tried changing CsvConfiguration object in Context property:
`csv.Context.CsvConfiguration.Prepare...`
and it doesn't seem to bee working as well

**Comment by JoshClose** [comment]
No it's essentially read only. All the values that the writer uses are copied local.
